# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_curl.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./lib/hostip.c.orig	Wed May  1 14:11:50 2002
+++ ./lib/hostip.c	Wed May  1 14:12:26 2002
@@ -405,7 +405,7 @@
   in.s_addr = htonl(num);
 
 #if defined(HAVE_INET_NTOA_R)
-  inet_ntoa_r(in,addr,addr_len);
+  inet_ntoa_r(in,addr);
 #else
   strncpy(addr,inet_ntoa(in),addr_len);
 #endif
--- ./lib/ftp.c.orig	Wed May  1 14:12:49 2002
+++ ./lib/ftp.c	Wed May  1 14:13:06 2002
@@ -939,7 +939,7 @@
   infof(conn->data, "Connecting to %s (%s) port %u\n",
         answer?answer->h_name:newhost,
 #if defined(HAVE_INET_NTOA_R)
-        inet_ntoa_r(in, ntoa_buf, sizeof(ntoa_buf)),
+        inet_ntoa_r(in, ntoa_buf),
 #else
         inet_ntoa(in),
 #endif
@@ -1283,7 +1283,7 @@
 #ifdef HAVE_INET_NTOA_R
     /* ignore the return code from inet_ntoa_r() as it is int or
        char * depending on system */
-    inet_ntoa_r(in, ntoa_buf, sizeof(ntoa_buf));
+    inet_ntoa_r(in, ntoa_buf);
     sscanf( ntoa_buf, "%hu.%hu.%hu.%hu",
             &ip[0], &ip[1], &ip[2], &ip[3]);
 #else
--- ./lib/url.c.orig	Wed May  1 14:13:24 2002
+++ ./lib/url.c	Wed May  1 14:13:33 2002
@@ -1307,7 +1307,7 @@
     (void) memcpy(&in.s_addr, &conn->serv_addr.sin_addr, sizeof (in.s_addr));
     infof(data, "Connected to %s (%s) port %d\n", conn->hostaddr->h_name,
 #if defined(HAVE_INET_NTOA_R)
-          inet_ntoa_r(in, ntoa_buf, sizeof(ntoa_buf)),
+          inet_ntoa_r(in, ntoa_buf),
 #else
           inet_ntoa(in),
 #endif
--- ./lib/if2ip.c.orig	Wed May  1 14:13:47 2002
+++ ./lib/if2ip.c	Wed May  1 14:13:54 2002
@@ -109,7 +109,7 @@
       struct sockaddr_in *s = (struct sockaddr_in *)&req.ifr_dstaddr;
       memcpy(&in, &(s->sin_addr.s_addr), sizeof(in));
 #if defined(HAVE_INET_NTOA_R)
-      ip = inet_ntoa_r(in,buf,buf_size);
+      ip = inet_ntoa_r(in,buf);
 #else
       ip = strncpy(buf,inet_ntoa(in),buf_size);
       ip[buf_size - 1] = 0;
